rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -946,32 +1032,30 @@ size_t Pir2Rir::compile(Context& ctx, Code* code) {
                 break;
             }
             case Tag::CallSafeBuiltin: {
-                assert(false && "not yet implemented.");
+                auto blt = CallSafeBuiltin::Cast(instr);
+                // no environment
+                blt->eachArg([&](Value* arg) { load(it, arg); });
+                cs.insertStackCall(Opcode::static_call_stack_, blt->nargs(), {},
+                                   Pool::get(blt->srcIdx), blt->blt);
+                store(it, blt);
                 break;
             }
             case Tag::MkEnv: {
                 auto mkenv = MkEnv::Cast(instr);
                 loadEnv(it, mkenv->parent());
-                cs << BC::makeEnv() << BC::dup() << BC::setEnv();
+                cs << BC::makeEnv() << BC::setEnv();
                 currentEnv = mkenv;
                 // bind all args
                 // TODO: maybe later have a separate instruction for this?
                 mkenv->eachLocalVar([&](SEXP name, Value* val) {
                     load(it, val);
                     cs << BC::stvar(name);
                 });
+                cs << BC::getEnv();
                 store(it, mkenv);
                 break;
             }
             case Tag::Phi: {
-                auto phi = Phi::Cast(instr);
-                phi->eachArg([&](BB*, Value* arg) {
-                    assert(((alloc.onStack(phi) && alloc.onStack(arg)) ||
-                            (alloc[phi] == alloc[arg])) &&
-                           "Phi inputs must all be allocated in 1 slot");
```

Comment:
why is this no longer true?

Reply:
It is but it is checked by `SSAAllocator::verify()`

Reply:
uhm, ok, but still, we probably want to run the verifier at the end of compilation, whereas the allocator verify is for debugging